nm-proxy/proxy/proxy.go

```diff
@@ -0,0 +1,271 @@
+package proxy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"runtime"
+
+	"github.com/gravitl/netclient/nm-proxy/common"
+	"github.com/gravitl/netclient/nm-proxy/config"
+	"github.com/gravitl/netclient/nm-proxy/models"
+	"github.com/gravitl/netmaker/logger"
+)
+
+// Proxy -  struct for wg proxy
+type Proxy struct {
+	Ctx        context.Context
+	Cancel     context.CancelFunc
+	Config     models.ProxyConfig
+	RemoteConn *net.UDPAddr
+	LocalConn  net.Conn
+}
+
+// Proxy.Start - starts proxying the peer
+func (p *Proxy) Start() error {
+
+	var err error
+	p.RemoteConn = p.Config.PeerEndpoint
+	logger.Log(0, "----> Established Remote Conn with RPeer: %s, ----> RAddr: %s", p.Config.RemoteKey.String(), p.RemoteConn.String())
+	addr, err := GetFreeIp(models.DefaultCIDR, p.Config.WgInterface.Device.ListenPort)
+	if err != nil {
+		logger.Log(1, "Failed to get freeIp: ", err.Error())
+		return err
+	}
+	wgListenAddr, err := GetInterfaceListenAddr(p.Config.WgInterface.Device.ListenPort)
+	if err != nil {
+		logger.Log(1, "failed to get wg listen addr: ", err.Error())
+		return err
+	}
+	if runtime.GOOS == "darwin" {
```

Comment:
can we put a comment here explaining why this is needed for an OS specific operation